test/e2e/clusterctl_upgrade_test.go

```diff
@@ -36,7 +36,8 @@ var _ = Describe("When testing clusterctl upgrades (v0.3=>current)", func() {
 			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.25/clusterctl-{OS}-{ARCH}",
 			InitWithProvidersContract: "v1alpha3",
 			// CAPI v0.3.x does not work on Kubernetes >= v1.22.
-			InitWithKubernetesVersion: "v1.21.12",
+			InitWithKubernetesVersion: "v1.21.14",
+			WorkloadKubernetesVersion: "v1.22.17",
```

Comment:
Very last comment: let's add a note to the v1.4=>v1.5 doc to tell providers that they should do the same
(in fact if they run v0.3 clusterctl upgrade tests they **have** to do the same)

Reply:
yes let's do it, but first i want to let the current test to finish (also, I wonder if there are providers still running 0.3=> current😄)

Reply:
Absolutely!

No idea to be honest. But the ones who look at core CAPI for guidance might still do.

Reply:
Done